includes/Modules/Thank_With_Google/Settings.php

```diff
@@ -69,4 +69,103 @@ protected function get_default() {
 			'buttonPostTypes' => array( 'post' ),
 		);
 	}
+
+	/**
+	 * Gets the setting value.
+	 *
+	 * @since m.e.x.t
+	 *
+	 * @return array
+	 */
+	public function get() {
+		$option = parent::get();
+
+		if ( is_array( $option ) ) {
+			if ( isset( $option['buttonPostTypes'] ) ) {
+				$option['buttonPostTypes'] = $this->sanitize_button_post_types( $option['buttonPostTypes'] );
+			}
+		}
+
+		return $option;
+	}
+
+	/**
+	 * Gets the callback for sanitizing the setting's value before saving.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable|null
+	 */
+	protected function get_sanitize_callback() {
+		return function( $option ) {
+			if ( is_array( $option ) ) {
+				if ( isset( $option['colorTheme'] ) ) {
+					$option['colorTheme'] = $this->sanitize_color_theme( $option['colorTheme'] );
+				}
+				if ( isset( $option['buttonPlacement'] ) ) {
+					$option['buttonPlacement'] = $this->sanitize_button_placement( $option['buttonPlacement'] );
+				}
+				if ( isset( $option['buttonPostTypes'] ) ) {
+					$option['buttonPostTypes'] = $this->sanitize_button_post_types( $option['buttonPostTypes'] );
+				}
+			}
+			return $option;
+		};
+	}
+
+	/**
+	 * Sanitizes the value of color theme.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $color_theme The value to sanitize.
+	 * @return string The sanitized value.
+	 */
+	private function sanitize_color_theme( $color_theme ) {
+		if ( in_array( $color_theme, array( 'blue' ), true ) ) {
+			return $color_theme;
+		}
+		return '';
+	}
+
+	/**
+	 * Sanitizes the value of button placement.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $button_placement The value to sanitize.
+	 * @return string The sanitized value.
+	 */
+	private function sanitize_button_placement( $button_placement ) {
+		if ( in_array(
+			$button_placement,
+			array(
+				'dynamic_low',
+				'dynamic_high',
+				'static_auto',
+				'static_above-content',
+				'static_below-content',
+				'static_below-first-paragraph',
+			),
+			true
+		) ) {
+			return $button_placement;
+		}
+		return '';
+	}
+
+	/**
+	 * Sanitizes the value of button post types.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $button_post_types The value to sanitize.
+	 * @return array The sanitized value.
+	 */
+	private function sanitize_button_post_types( $button_post_types ) {
+		if ( ! is_array( $button_post_types ) ) {
+			return array( 'post' );
+		}
+		return array_intersect( $button_post_types, get_post_types( array( 'public' => true ) ) );
```

Comment:
Additionally, I wonder if this matches the UI we show… in the UI we only show posts that support `viewable` (see: https://github.com/google/site-kit-wp/issues/5456). I wonder if this would allow a user to submit a post type successfully that wouldn't appear in the UI.

Not a big deal as they'd need to intentionally do so, but it would be nice if the only posts we allowed to save would be ones that would appear with the `'viewable'` value in `supports`. 🤔 

I know this is what was in the ACs/IB, but I think it's maybe a bit inaccurate? Would there be an easy way to get all post types that match what `wp.data.select('core').getPostTypes().filter((postType) => postType.viewable)` returns? If so let's do that 😄

Reply:
Hi, the API for `getPostTypes()` uses `is_post_type_viewable` for each post to determine the viewable status. For a post to be viewable: 

* if it's a builtin post type, the `public` must be `true`.
* If it's not a builtin post type, the `publicly_queryable` must be `true`. 

Which definitely makes your point valid. I'll see what's the best way to achieve that here. 

Off Topic: speaking of `getPostTypes` be aware that it only returns 10 post types if the site has more. See [here](https://github.com/WordPress/gutenberg/issues/15413). The workaround is to supply `{ per_page: -1 }` to the selector which will return 100. If there's 101 post type this workaround will not work :grin:

Reply:
Yeah, sadly I think that's our best option for now until someone runs into an issue 😅 

We could add pagination if needed, but for now `101` is probably enough.